go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -346,6 +349,7 @@ func (bs *S3BackupStorage) client() (*s3.S3, error) {
 				LogLevel:         logLevel,
 				Endpoint:         aws.String(*endpoint),
 				Region:           aws.String(*region),
+				MaxRetries:       aws.Int(*retryCount),
```

Comment:
Although we don't need tests for something like this, I also think we shouldn't override the defaults unless the user specifies a parameter.

So, a better change would be to make the flag default to -1 (meaning the value is unset) and then only set the value in the config if it's 0 or greater.

That way we're not changing existing behavior in potentially subtle ways.

Reply:
Ah this is a good callout. @RicardoLorenzo, here an example of how we have done that in the past: https://github.com/vitessio/vitess/pull/4037

Reply:
I'm sorry, I was sick last week and just seeing the comments now. I will look into this.

Reply:
I have implemented the proposed change.